### PR TITLE
Complete Lab 5

### PR DESCRIPTION
**Pull request by shampa077**
Fixes #

Review of colleague's PR #

### CODE REVIEW

File: 05_stringer/shahriys/main.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var outstringer io.Writer = os.Stdout
```
Comment (lexual) [inline 05_stringer/shahriys/main.go:9]: if the variable name is meant to mean "out stringer", I believe CamelCase is preferred in go code. i.e. outStringer.

Alternatively just plain old 'out' might be fine too

File: 05_stringer/shahriys/main.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+type IPAddr [4]byte
+
+// String satisfies the fmt.Stringer interface for the IPAddr type
+func (u IPAddr) String() string {
```
Comment (lexual) [inline 05_stringer/shahriys/main.go:14]: would 'ip' be a better variable name to use than 'u' ?

File: 05_stringer/shahriys/main_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	var bufstringer bytes.Buffer
+	outstringer = &bufstringer
```
Comment (lexual) [inline 05_stringer/shahriys/main_test.go:12]: see above comment about variable names & camelCase. relevant for bufstringer and outstringer

File: 05_stringer/shahriys/main_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		expected string
+	}
+
+	tests := []test{
+		{input: IPAddr{127, 0, 0, 1}, expected: `127.0.0.1`},
+		{input: IPAddr{}, expected: `0.0.0.0`},
+	}
+
+	for _, tc := range tests {
+		a := tc.input
```
Comment (lexual) [inline 05_stringer/shahriys/main_test.go:36]: is 'a' variable necessary.

Perhaps the next line could just become:

```
actual := tc.input.String()
```

File: 05_stringer/shahriys/main_test.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+		actual := a.String()
+		if !reflect.DeepEqual(tc.expected, actual) {
```
Comment (lexual) [inline 05_stringer/shahriys/main_test.go:38]: if we are just comparing strings, I think you can just do:

```if tc.expected != actual```

**Comment by juliaogris** [comment]
The go-course is now closed. Thank you very much for participating.

**Pull request closed without merging**